fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/FedoraResourceImpl.java

```diff
@@ -0,0 +1,262 @@
+    public ResourceHeaders getHeaders() {
+        return headers;
+    }
+
+    @Override
+    public void setHeaders(final ResourceHeaders headers) {
+        this.headers = headers;
+    }
+
+    @Override
+    public Stream<FedoraResource> getChildren(final Boolean recursive) {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public FedoraResource getContainer() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public FedoraResource getOriginalResource() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public FedoraResource getTimeMap() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public Instant getMementoDatetime() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public boolean isMemento() {
+        // TODO Auto-generated method stub
+        return false;
+    }
+
+    @Override
+    public boolean isAcl() {
+        // TODO Auto-generated method stub
+        return false;
+    }
+
+    @Override
+    public FedoraResource findMementoByDatetime(final Instant mementoDatetime) {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public FedoraResource getAcl() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public FedoraResource findOrCreateAcl() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public FedoraResource getChild(final String relPath) {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public boolean hasProperty(final String relPath) {
+        // TODO Auto-generated method stub
+        return false;
+    }
+
+    @Override
+    public void delete() {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public Instant getCreatedDate() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public Instant getLastModifiedDate() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public boolean hasType(final String type) {
+        // TODO Auto-generated method stub
+        return false;
+    }
+
+    @Override
+    public List<URI> getTypes() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public void addType(final String type) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public RdfStream getTriples() {
+        if (triplesStream == null) {
+            try {
+                pSessionFactory.getSession(tx.getId())
+                        .readTriples(this);
+            } catch (final PersistentItemNotFoundException e) {
+                throw new ItemNotFoundException("Unable to retrieve triples for " + getId(), e);
```

Comment:
Maybe this should react differently if you have not persisted the FedoraResource yet? Like if you called create and then called getTriples() then it probably shouldn't check the storage. 
```
/**
 * Is this a brand new resource?
 */
private boolean isNew = true;

public RdfStream getTriples() {
    if (triplesStream == null && !isNew) {
            try {
                this.triplesStream = pSessionFactory.getSession(tx.getId()).readTriples(this);
            } catch (final PersistentItemNotFoundException e) {
                throw new ItemNotFoundException("Unable to retrieve triples for " + getId(), e);
            }
    }
    return triplesStream;
}
```

Reply:
Good call, that makes sense, I can add that in soon.